File: src/main/java/resort_management/entity/Payment.java
package resort_management.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.*;
import resort_management.enums.PaymentMethod;
import resort_management.enums.PaymentStatus;
import com.fasterxml.jackson.annotation.JsonIgnore;
import java.time.LocalDateTime;
import java.math.BigDecimal;

@Entity
@Table(name = "payments")
@Data
@EqualsAndHashCode(callSuper = true, exclude = {"booking"})
@NoArgsConstructor
@AllArgsConstructor
public class Payment extends BaseEntity {

    @OneToOne(fetch = FetchType.LAZY) // ĐÃ SỬA: Chuyển sang LAZY
    @JoinColumn(name = "booking_id", nullable = false, unique = true)
    @JsonIgnore
    @ToString.Exclude
    private Booking booking;

    @PositiveOrZero(message = "Số tiền không được âm")
    @Column(nullable = false, precision = 18, scale = 2)
    private BigDecimal amount;

    @Column(name = "discount_amount", precision = 18, scale = 2)
    private BigDecimal discountAmount = BigDecimal.ZERO;

    @Enumerated(EnumType.STRING)
    @Column(name = "payment_method", length = 10, nullable = false)
    private PaymentMethod paymentMethod = PaymentMethod.CASH;

    @Enumerated(EnumType.STRING)
    @Column(name = "payment_status", length = 20, nullable = false)
    private PaymentStatus paymentStatus = PaymentStatus.PENDING;

    @Column(name = "payment_date")
    private LocalDateTime paymentDate;
}

File: frontend/src/types/payment.ts
import { BookingDetail } from "./booking";

export interface Payment {
  id: number;
  bookingId: number;
  amount: number;
  discountAmount: number; // ← thêm vào đây
  paymentMethod: string;
  paymentStatus: string;
  paymentDate: string | null;
  createdAt: string;
  bookingDetail?: BookingDetail;
}

export interface PaymentPageData {
  content: Payment[];
  page: number;
  totalPages: number;
  totalElements: number;
  last: boolean;
}

export interface RevenueItem {
  period: string;
  revenue: number;
}
